### PR TITLE
update tutorials for 0.4 (#223)

### PR DESCRIPTION
**Pull request by soumith**
* Update beginner tutorials

* Update rest tutorials

* update q learning

* format

* rewrote part of fast neural transfer to add normalization and let it make much more sense

* improve fast neural transfer code again

### CODE REVIEW

File: beginner_source/former_torchies_tutorial.rst [hunk 1]
```diff
@@ -25,9 +25,6 @@ In this tutorial, you will learn the following:
 .. galleryitem:: /beginner/former_torchies/tensor_tutorial.py
     :figure: /_static/img/tensor_illustration_flat.png
 
-.. galleryitem:: /beginner/former_torchies/autograd_tutorial.py
```
Comment (chsasank) [inline beginner_source/former_torchies_tutorial.rst]: @soumith Why is this removed? former_torchies/autograd_tutorial.py itself is edited as a part of the PR. Should I add this back?

Comment (soumith) [reply beginner_source/former_torchies_tutorial.rst]: yes please, add this back. Not sure what happened here. cc: @SsnL

Comment (soumith) [reply beginner_source/former_torchies_tutorial.rst]: added back now.

Comment (chsasank) [reply beginner_source/former_torchies_tutorial.rst]: thanks! Will build the latest docs.
